### PR TITLE
Milestone 10: Feature authorization

### PR DESCRIPTION
**Pull request by darikmohammed**
# Feature 
- Install CanCanCan
- Add a role column to the users table.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin") using CanCanCan.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin") using CanCanCan.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -25,4 +25,11 @@ def create
       redirect_to new_user_post_path
     end
   end
+
```
Comment (danmainah) [inline app/controllers/posts_controller.rb:28]: - [ ] Kindly add  CanCanCan for Post authorization, this is a project requirement

Comment (danmainah) [reply app/controllers/posts_controller.rb:28]: I am also running into this error when I run your app:
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @darikmohammed and @isaacpitwa , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (darikmohammed) [reply app/controllers/posts_controller.rb:28]: @danmainah your request is not clear can you please elaborate on it? 

- I have added CanCanCan for authorization to delete a post and delete a comment. You can see it in the [ability.rb](https://github.com/darikmohammed/blog-app/pull/10/files#diff-0fdc2ce2595c4fe479d6a97b404013651e8a0e2d4b02d93c87843b3a30103d9f)

- I am using Tailwind CSS to style my application and it is working fine on my end.

I have shared the work of my project in the video below.

https://user-images.githubusercontent.com/56404835/193068869-29a2849c-fd24-4f44-8554-07f085e71706.mov

**Review by m05e5 — APPROVED** [review]
# Status: ✔️ Project Approved ✔️ 

Hello @darikmohammed   🔥 

Good job on implementing the requirements. You smashed it. Well Done.
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

goodjob


Now you can merge this branch 🚀 

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

_m05e5_

### 💻 👩🏽‍💻  Happy Coding!  👨🏽‍💻 🖥️

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**